### PR TITLE
REV-146 my listings page bugs

### PR DESCRIPTION
**Pull request by IbrahimAljabrST**
### Jira URL

https://optimumpartners.atlassian.net/browse/REV-146

### Description

my listings page bugs

### Dependencies in Other PRs,Repo

https://github.com/abukhadraop-or/reverifi-api/pull/38

### CODE REVIEW

File: src/components/my-listings/ListingsHeader/index.jsx [hunk 1]
```diff
@@ -1,36 +1,55 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/prop-types */
```
Comment (mohdabed123) [inline src/components/my-listings/ListingsHeader/index.jsx:2]:
```suggestion
```

File: src/components/my-listings/ListingsHeader/index.jsx [hunk 2]
```diff
@@ -1,36 +1,55 @@
...
-  FilterIconContainer,
   Header,
-  HeaderTitle,
 } from './listings-header.style';
 
 /**
  * My Profile page component.
  *
  * @return {JSX.Element}
  */
```
Comment (mohdabed123) [inline src/components/my-listings/ListingsHeader/index.jsx:19]: update docs

File: src/components/my-listings/ListingsTable/index.jsx [hunk 3]
```diff
@@ -1,22 +1,22 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable no-unused-vars */
```
Comment (mohdabed123) [inline src/components/my-listings/ListingsTable/index.jsx:2]:
```suggestion
```

File: src/components/my-listings/ListingsTable/index.jsx [hunk 4]
```diff
@@ -34,12 +35,15 @@ const tableHeaders = ['IMAGE', 'PROPERTY', 'SELLER', 'DATE', 'STATUS', ''];
  *
```
Comment (mohdabed123) [inline src/components/my-listings/ListingsTable/index.jsx:35]: update docs

File: src/components/my-listings/ListingsTable/index.jsx [hunk 5]
```diff
@@ -48,58 +52,29 @@ export default function MyListings({ setDeleteId }) {
...
-  }, [authInfo, pageNumber, showModal]);
-
   return (
     <>
       <Table headers={tableHeaders}>
         {listings?.data?.map(
           ({ address, images, agent, createdAt, id, owner }) => (
             <TableRow key={id}>
-              <TableCell>
+              <TableCellContainer>
```
Comment (mohdabed123) [inline src/components/my-listings/ListingsTable/index.jsx:61]:
```suggestion
              <CellContainer>
```

File: src/components/my-listings/ListingsTable/index.jsx [hunk 6]
```diff
@@ -161,8 +136,12 @@ export default function MyListings({ setDeleteId }) {
         <CurrentListing>{startItem} -</CurrentListing>
         <CurrentListing>{endItem}</CurrentListing>
         <MaxListingNumber>of {listings?.count}</MaxListingNumber>
-        <ArrowLeft onClick={handleLeftArrowClick} />
-        <ArrowRight onClick={handleRightArrowClick} />
+        <ArrowIconContainer onClick={handleLeftArrowClick}>
```
Comment (mohdabed123) [inline src/components/my-listings/ListingsTable/index.jsx:139]:
```suggestion
        <IconContainer onClick={handleLeftArrowClick}>
```

File: src/components/my-listings/MyListings/index.jsx [hunk 7]
```diff
@@ -1,26 +1,100 @@
...
+  };
+
+  const handleLeftArrowClick = () => {
+    if (pageNumber >= 1) {
+      setPageNumber(pageNumber - 1);
+    }
+  };
+
+  const handleRightArrowClick = () => {
+    if (pageNumber < Math.floor(listings?.count / listingPerPage)) {
```
Comment (mohdabed123) [inline src/components/my-listings/MyListings/index.jsx:53]:
```suggestion
    if (pageNumber < Math.trunc(listings?.count / listingPerPage)) {
```

File: src/components/my-listings/MyListings/index.jsx [hunk 8]
```diff
@@ -1,26 +1,100 @@
...
+  }, [authInfo, pageNumber, showModal, order]);
+
   return (
     <MyListingContainer>
-      <MyListingHeader />
-      <ShowModalProvider>
-        <ListingsTable setDeleteId={setDeleteId} />
-        <DeleteBox deleteId={deleteId} />
-      </ShowModalProvider>
+      {listings?.count > 0 ? (
```
Comment (mohdabed123) [inline src/components/my-listings/MyListings/index.jsx:70]:
```suggestion
      {listings? ? (
```

**Review by mohdabed123 — CHANGES_REQUESTED** [review]

File: src/components/my-listings/MyListings/index.jsx (hunk 8 above)
Comment (IbrahimAljabrST) [reply src/components/my-listings/MyListings/index.jsx:70]: this will return true without the .count

**Review by mohdabed123 — APPROVED** [review]

**Pull request merged**